src/kubelet/metric/metric.go

```diff
@@ -214,38 +211,31 @@ func GroupStatsSummary(statsSummary v1.Summary) (definition.RawGroups, []error)
 		return g, errs
 	}
 
-PodListLoop:
 	for _, pod := range statsSummary.Pods {
 		rawPodMetrics, rawEntityID, err := fetchPodStats(pod)
 		if err != nil {
 			errs = append(errs, err)
-			continue PodListLoop
+			continue
 		}
+
 		g["pod"][rawEntityID] = rawPodMetrics
-	VolumeListLoop:
 		for _, volume := range pod.VolumeStats {
 			rawVolumeMetrics, err := fetchVolumeStats(volume)
 			if err != nil {
 				errs = append(errs, err)
-				continue VolumeListLoop
+				continue
 			}
 			rawVolumeMetrics["podName"] = rawPodMetrics["podName"]
 			rawVolumeMetrics["namespace"] = rawPodMetrics["namespace"]
 			rawEntityID = fmt.Sprintf("%s_%s_%s", rawPodMetrics["namespace"], rawPodMetrics["podName"], rawVolumeMetrics["volumeName"])
 			g["volume"][rawEntityID] = rawVolumeMetrics
 		}
 
-		if pod.Containers == nil {
-			// Some pods could have no containers yet or containers could be in a back-off pulling image status.
-			continue PodListLoop
-		}
-
-	ContainerListLoop:
 		for _, container := range pod.Containers {
 			rawContainerMetrics, err := fetchContainerStats(container)
 			if err != nil {
 				errs = append(errs, err)
-				continue ContainerListLoop
+				continue
```

Comment:
Same as above, an error in a container will skip metrics for the whole pod.

Reply:
Will mark as duplicate.